explore/explore_internal.h

```diff
@@ -257,19 +257,23 @@ namespace exploration
     if (draw > total) //make very sure that draw can not be greater than total.
       draw = total;
 
+    bool index_found = false; //found chosen action
     float sum = 0.f;
     uint32_t i = 0;
     for (It pdf = pdf_first; pdf != pdf_last; ++pdf, ++i)
     {
       sum += *pdf;
-      if (sum > draw)
+      if (!index_found && sum > draw)
       {
         chosen_index = i;
-        return S_EXPLORATION_OK;
+        index_found = true;
       }
+      *pdf /= total;
     }
 
-    chosen_index = i - 1;
+    if(!index_found)
+      chosen_index = i - 1;
+    
```

Comment:
nit: whitespace

Reply:
fixed

---
In reply to: [228988454](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1658#discussion_r228988454) [](ancestors = 228988454)